test/WebSites/InlineConstraintsWebSite/Constraints/IsbnDigitScheme10Constraint.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Routing;
+
+namespace InlineConstraintsWebSite.Constraints
+{
+    public class IsbnDigitScheme10Constraint : IRouteConstraint
+    {
+        private readonly bool _allowDashes;
+
+        public IsbnDigitScheme10Constraint(bool allowDashes)
+        {
+            _allowDashes = allowDashes;
+        }
+
+        public bool Match(
+            HttpContext httpContext,
+            IRouter route,
+            string routeKey,
+            IDictionary<string, object> values,
+            RouteDirection routeDirection)
+        {
+            var isbnRegExpression = @"^(\d{9})(\d|X)\z";
+
+            if (_allowDashes)
+            {
+                isbnRegExpression = @"^(\d{1})[-]\d{3}[-]\d{5}[-][\d|X]\z";
+            }
+
+            object value;
+
+            if (!values.TryGetValue(routeKey, out value))
+            {
+                return false;
+            }
+
+            var isbnNumber = value as string;
+            var isbnRegEx = new Regex(isbnRegExpression, RegexOptions.IgnoreCase);
```

Comment:
Because there are only ever two regexes and they're both constant, can just store both variations statically as Regex objects and just choose which one to use at runtime. Also, I think you can just call `.IsMatch()` instead of calling `Match(...).Success`.